### ISSUE TITLE
"Current" version fails to retain prefs

### ISSUE THREAD

**Issue by Telmesomething** [MrAlex94/Waterfox#1236]
After spending about an hour configuring a fresh "Current" installation it failed to retain my prefs and everything has reverted to default again after being shutdown. This means I've got the (ugly) dark theme back again and the default settings some of which I don't want such as "Recommend extensions as you browse" and "Recommend features as you browse" have been re-enabled. Easy enough to switch to a light theme again I know,  but the new browser should respect a user's settings and not take it upon itself to undo all the hard work. 

In addition I'd installed 23 extensions of which only two remain. Also, one extension which I didn't install namely Video Download Helper 7.3.6 has been installed, but disabled. Here's a screenshot:

![Extensions removed](https://user-images.githubusercontent.com/33827636/67635553-e6276580-f8c8-11e9-9da3-501a7e9dc7f9.jpg)

On the other hand all the about:config changes I made have been retained so at least I don't have to trawl through that lot again. 

But obviously the other mishaps need to be addressed a.s.a.p.

My system: 
Windows 8.1 Home (fully patched) running on an Intel i5 CPU. 
Laptop is an Acer Aspire V3-772G purchased in February 2014.
It's also an Optimus system which means it contains two video cards. The Intel 4600 handles menial tasks which don't require a load of graphics processing power while the Nvidia GT 750M kicks in automatically when required.

**Comment by laniakea64** [comment]
Could Waterfox have switched profiles on you?  What do you have in `about:profiles`?

**Comment by grahamperrin** [comment]
Did you use both the soft launch and hard launch builds of Waterfox Current 2019.10?

The soft was the installer that disappeared. The hard was the identically named installer for the same version that later appeared.

Also please visit: 

- about:profiles

What are the names of the profiles?

**Comment by Telmesomething** [comment]
I updated Classic using the About:Firefox option since auto-update has been disabled (by me). I subsequently downloaded Current via the releases site.

This morning I totally gave up on Current and decided to uninstall it. Unfortunately there's seems to have been some interaction between Current and Classic whereby the add-ons I had installed in Classic were removed as well along with the customizations I'd created. In that respect I have four profiles which I use. I keep one solely for online banking, another one for a specific site I buy a lot of stuff from (Zalando.nl), the 3rd one for online tax returns and the 4th for general purposes. This latter one survived intact while the others were all returned to default settings. The one used for online banking could no longer login anymore. The profile for Zalando couldn't login either. 

My solution so far has been to run Windows System Restore several times until finding one which works. Oddly enough, after restoring to a manual point prior to installing Current those settings appeared in About Waterfox Classic i.e. 2019.10. I can't figure out how that could happen bearing in mind that Current hadn't been installed at that point (?).

After running SR once again WF in now back to 56.2.13 and I can login to my bank a/c again. 

![after_SR](https://user-images.githubusercontent.com/33827636/67679587-b12f1780-f989-11e9-94fa-5d8d9193db17.jpg)

I've been through the registry, but can't find any settings for Current so I presume it's gone.

Since I can't afford not to have access to my bank a/c I'm going to leave Current well alone until a stable version is released. 

Many thanks anyway to those who took the time to try and help.

**Comment by Telmesomething** [comment]
Just noticed a request for build ID for another user so thought I would post that here in case it helps: Build ID 20190902231505

**Comment by grahamperrin** [comment]
@Telmesomething big thanks for beta testing Waterfox Current. 

Now, reading between the lines. 

With Waterfox Classic (56.2.14 or 2019.10): 

- at any time, did you use **Waterfox Profile Manager**? 

Like this, but with different names for your profiles: 

![image](https://user-images.githubusercontent.com/192271/67824145-7cbc7800-fabd-11e9-99ad-60c1cb0ada7c.png)

**Comment by Telmesomething** [comment]
Hi Graham,

Yes I use the Profile Manager all the time using different names to identify each of them. They're called respectively "Waternet" (used originally to login to the local council utility to pay my water bill, but has now become my main profile), "ING" (bank account), "22-January-2019" (old profile I don't use anymore), "Zalando" (used for buying stuff from that company based in Germany) and "NOS WK" to watch football on Dutch TV. With the exception of Classic Theme Restorer used to place tabs below the location bar the only profile which has a large number of addons installed is the first one (Waternet). WF is configured to delete all the options such as browsing history, cache, cookies etc., in all the other three profiles. In that respect it's impossible to login to any of the sites the profiles have been created for unless third party cookies, trackers etc., are enabled. My solution to that is to delete everything when I close those particular profiles. 

When I first installed Current it appears to have created a new profile without a description i.e. there was just a blank line in the Profile Manager, but the Classic profiles were all visible. 

I have a peculiar problem with Zalando in that I can't login even though I use a password manager (Keepass) to store and copy/paste passwords. The error message in the image translated says "_Sorry, something went wrong, but we're working on it. Try again later_". Zalando support's only solution is to ask me to change my password again. I've done that 13 times already since the beginning of the year. I've already verified that the password length and characters used are entirely up to me. 

![GITHUB](https://user-images.githubusercontent.com/33827636/67845584-02f9ae00-fb00-11e9-9772-3fb9c01f2547.jpg)

**Comment by grahamperrin** [comment]
Thanks, 

> When I first installed Current it appears to have created a new profile without a description i.e. there was just a blank line in the Profile Manager, 

– I haven't seen any other comparable report. 

Please: did you ever use Waterfox Current Waterfox Profile Manager to open any of the four Waterfox Classic profiles (_Waternet_, _ING_, _22-January-2019_, _Zalando_)? 

Vice versa, did you at any time use Waterfox Classic Waterfox Profile Manager to open the one Waterfox Current profile?

**Comment by Telmesomething** [comment]
> Please: did you ever use Waterfox Current Waterfox Profile Manager to open any of the four Waterfox Classic profiles 

No, to both questions.

**Comment by grahamperrin** [comment]
> No, to both questions.

🥇 for taking such great care. 

So. Other possible explanations for the symptoms in your case … 

> … I didn't install … Video Download Helper 7.3.6 … installed, but disabled. 

Video DownloadHelper version history https://addons.mozilla.org/addon/video-downloadhelper/versions/ all versions require Firefox Quantum i.e. 57.⋯ and the 57 pretence in 56-based Waterfox Classic means that you might have installed Version Video DownloadHelper 7.0.0 or later in one of your four profiles for Waterfox Classic. 

When I last asked, the import/migration routine was not yet automated for Waterfox Current. @MrAlex94 please, is that still true? Or does the wizard start automatically, since 2019.10 and 2019.10 use a single `profiles.ini` file?

**Comment by Telmesomething** [comment]
> Video DownloadHelper version history https://addons.mozilla.org/addon/video-downloadhelper/versions/ all versions require Firefox Quantum i.e. 57.⋯ and the 57 pretence in 56-based Waterfox Classic means that you might have installed Version Video DownloadHelper 7.0.0 or later in one of your four profiles for Waterfox Classic.

That's correct. I have video-downloadhelper 6.3.3. installed in my main profile. I would have thought though that the Current installation would be a barebones affair meaning that no addons at all should have been installed. The fact that 7.3.6 was installed would seem to indicate that the Current version peruses the Classic installation extensions folder and installs addons which would be compliant with the new version. If it does that does it also copy about:config prefs which the user has enabled in the Classic version and enables them in Current? If so could any of them interfere with the operation of the new version?

That still doesn't explain why Current would choose to remove 21 addons which had been installed via Mozilla AMO. Similarly, why were defaults re-enabled including "Recommend features as you browse"  for example. I forgot to mention before that the default search engines were reinstated along with the other default settings. I'd only installed one search engine myself which was Startpage.com. But that had been deleted as well. 

One other thing I didn't like was the "Visit" and "Search with" being present in Current. Those two settings are especially irritating. In that respect surely MrAlex94 doesn't have to follow everything Mozilla does?

**Comment by grahamperrin** [comment]
>> … I'd installed 23 extensions of which only two remain. …

> … why Current would choose to remove 21 addons …

Are the 23 still in your _main_ profile in Waterfox Classic? If so, please let us have troubleshooting information from that profile. 

With Waterfox Classic running and using the 23-add-on profile, please use the **Copy text to clipboard** button at: 

- about:support

### Browser identification

From <https://github.com/MrAlex94/Waterfox/issues/1236#issuecomment-546936157> above: 

> 20190902231505

That's not a match for anything at or near https://github.com/grahamperrin/Waterfox/wiki/Archive-of-change-logs,-announcements-and-downloads#201910-current and – critically – given the September date of the build, it can not be any 2019.10 release of Waterfox Current. 

Your opening post mentioned a _fresh "Current" installation_, now I wonder whether it was truly fresh at the time. It's possible/likely that an outdated installer <strike>or installation </strike>was used. What was its origin?

**Comment by Telmesomething** [comment]
As regards the installer I downloaded it from the Waterfox blog/Releases page after reading about it in this [ghacks.net](https://www.ghacks.net/2019/10/25/waterfox-development-splits-into-classic-and-current-branches/) article. Since the article is dated 25/10 the download must have taken place sometime after that. 

However, I wasn't paying attention at the time and inadvertently downloaded the Classic installer instead of the Current one. Needless to say after running that I lost all my profiles. That led me to running Win 8.1 System Restore several times until 2019.10 didn't reappear in the "About Waterfox" menu. 

It wasn't until several days later that I went back to the Releases page and scrolled further down the page that I discovered the Current installer. I assume it was up to date but I've no way of knowing if that's true or not I'm afraid.

Last but not least here's the troubleshooting info you requested. 

``Application Basics
------------------

Name: Waterfox
Version: 2019.10
Build ID: 20191022125504
Update Channel: release
User Agent: Mozilla/5.0 (Windows NT 6.3; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0 Waterfox/56.3
OS: Windows_NT 6.3
Multiprocess Windows: 0/1 (Disabled)
Web Content Processes:
Stylo: false (disabled by default)
Google Key: Missing
Mozilla Location Service Key: Missing
Safe Mode: false

Waterfox Features
-----------------

Extensions
----------

Name: Absolute Enable Right Click & Copy
Version: 1.3.8
Enabled: true
ID: {9350bc42-47fb-4598-ae0f-825e3dd9ceba}

Name: Add to Search Bar
Version: 2.9
Enabled: true
ID: add-to-searchbox@maltekraus.de

Name: Behind The Overlay
Version: 0.1.6
Enabled: true
ID: jid1-Y3WfE7td45aWDw@jetpack

Name: CanvasBlocker
Version: 0.4.4b
Enabled: true
ID: CanvasBlocker@kkapsner.de

Name: Classic Theme Restorer
Version: 1.7.7.2
Enabled: true
ID: ClassicThemeRestorer@ArisT2Noia4dev

Name: Close Tab By Double Click
Version: 1.14.1-signed.1-signed
Enabled: true
ID: close@doubleclick

Name: Context Search
Version: 0.6.4
Enabled: true
ID: {902D2C4A-457A-4EF9-AD43-7014562929FF}

Name: CSS Exfil Protection
Version: 1.0.17
Enabled: true
ID: {7fc8ef53-24ec-4205-87a4-1e745953bb0d}

Name: Decentraleyes
Version: 2.0.13
Enabled: true
ID: jid1-BoFifL9Vbdl2zQ@jetpack

Name: google redirect rewrite remover
Version: 0.7
Enabled: true
ID: jid1-29GIO3byutyhpw@jetpack

Name: Google search link fix
Version: 1.6.8
Enabled: true
ID: jid0-XWJxt5VvCXkKzQK99PhZqAn7Xbg@jetpack

Name: HTTPS Everywhere
Version: 2019.6.4
Enabled: true
ID: https-everywhere@eff.org

Name: I don't care about cookies
Version: 3.0.5
Enabled: true
ID: jid1-KKzOGWgsW3Ao4Q@jetpack

Name: Image Max URL
Version: 0.11.9
Enabled: true
ID: maxurl@qsniyg

Name: Legibility
Version: 2.0.4
Enabled: true
ID: {4fe52e6f-3e1a-471a-a32f-193dbc3c647d}

Name: No Coin - Block miners on the web!
Version: 0.4.13
Enabled: true
ID: {5657c026-efc3-4860-b43b-16e4eaa8a9aa}

Name: No Resource URI Leak
Version: 1.1.0
Enabled: true
ID: @no-resource-uri-leak

Name: Privacy Badger
Version: 2019.10.28
Enabled: true
ID: jid1-MnnxcxisBPnSXQ@jetpack

Name: ReminderFox
Version: 2.1.6.3
Enabled: true
ID: {ada4b710-8346-4b82-8199-5de2b400a6ae}

Name: Self-Destructing Cookies
Version: 0.4.12
Enabled: true
ID: jid0-9XfBwUWnvPx4wWsfBWMCm4Jj69E@jetpack

Name: Status-4-Evar
Version: 2017.08.20.15
Enabled: true
ID: status4evar@caligonstudios.com

Name: Thumbnail Zoom Plus
Version: 4.2
Enabled: true
ID: thumbnailZoom@dadler.github.com

Name: uBlock Origin
Version: 1.23.0
Enabled: true
ID: uBlock0@raymondhill.net

Name: Classic Add-ons Archive
Version: 2.0.1
Enabled: false
ID: ca-archive@Off.JustOff

Name: User-Agent Switcher
Version: 1.3.15
Enabled: false
ID: user-agent-switcher@ninetailed.ninja

Name: Video DownloadHelper
Version: 6.3.3
Enabled: false
ID: {b9db16a4-6edc-47ec-a1f4-b86292ed211d}

Graphics
--------

Features
Compositing: Basic
Asynchronous Pan/Zoom: none
WebGL 1 Driver WSI Info: -
WebGL 1 Driver Renderer: WebGL is currently disabled.
WebGL 1 Driver Version: -
WebGL 1 Driver Extensions: -
WebGL 1 Extensions: -
WebGL 2 Driver WSI Info: -
WebGL 2 Driver Renderer: WebGL is currently disabled.
WebGL 2 Driver Version: -
WebGL 2 Driver Extensions: -
WebGL 2 Extensions: -
DirectWrite: true (6.3.9600.19457)
GPU #1
Active: Yes
Description: Intel(R) HD Graphics 4600
Vendor ID: 0x8086
Device ID: 0x0416
Driver Version: 10.18.14.5057
Driver Date: 8-16-2018
Drivers: igdumdim64 igd10iumd64 igd10iumd64 igdumdim32 igd10iumd32 igd10iumd32
Subsys ID: 07811025
RAM: Unknown
GPU #2
Active: No
Description: NVIDIA GeForce GT 750M
Vendor ID: 0x10de
Device ID: 0x0fe4
Driver Version: 25.21.14.1917
Driver Date: 2-20-2019
Drivers: nvd3dumx,nvwgf2umx,nvwgf2umx nvd3dum,nvwgf2um,nvwgf2um
Subsys ID: 07811025
RAM: 4096
Diagnostics
ClearType Parameters: Gamma: 2.2 Pixel Structure: RGB ClearType Level: 100 Enhanced Contrast: 400
AzureCanvasAccelerated: 0
AzureCanvasBackend: skia
AzureContentBackend: skia
AzureFallbackCanvasBackend: cairo
ClearType Parameters: Gamma: 2.2 Pixel Structure: RGB ClearType Level: 100 Enhanced Contrast: 400
Decision Log
HW_COMPOSITING:
disabled by user: Disabled by pref
D3D11_COMPOSITING:
unavailable by default: Hardware compositing is disabled
DIRECT2D:
unavailable by default: Direct2D requires Direct3D 11 compositing
D3D11_HW_ANGLE:
unavailable by default: D3D11 compositing is disabled
disabled by env: D3D11 compositing is disabled
GPU_PROCESS:
unavailable by default: Multi-process mode is not enabled
WEBRENDER:
opt-in by default: WebRender is an opt-in feature
unavailable by runtime: ANGLE is disabled




Media
-----

Audio Backend: wasapi
Max Channels: 2
Preferred Channel Layout: stereo
Preferred Sample Rate: 48000
Output Devices
Name: Group
Speakers (Realtek High Definition Audio): HDAUDIO\FUNC_01&VEN_10EC&DEV_0282&SUBSYS_10250781&REV_1000\4&2e4824cd&0&0001
Input Devices
Name: Group
Stereo Mix (Realtek High Definition Audio): HDAUDIO\FUNC_01&VEN_10EC&DEV_0282&SUBSYS_10250781&REV_1000\4&2e4824cd&0&0001
Microphone (Realtek High Definition Audio): HDAUDIO\FUNC_01&VEN_10EC&DEV_0282&SUBSYS_10250781&REV_1000\4&2e4824cd&0&0001
Microphone (Realtek High Definition Audio): HDAUDIO\FUNC_01&VEN_10EC&DEV_0282&SUBSYS_10250781&REV_1000\4&2e4824cd&0&0001
Microphone (Realtek High Definition Audio): HDAUDIO\FUNC_01&VEN_10EC&DEV_0282&SUBSYS_10250781&REV_1000\4&2e4824cd&0&0001

Important Modified Preferences
------------------------------

accessibility.typeaheadfind.flashBar: 0
browser.cache.disk.capacity: 358400
browser.cache.disk.filesystem_reported: 1
browser.cache.disk.smart_size.first_run: false
browser.cache.frecency_experiment: 1
browser.display.use_document_fonts: 0
browser.download.useDownloadDir: false
browser.places.smartBookmarksVersion: 8
browser.search.suggest.enabled: false
browser.search.useDBForOrder: true
browser.sessionstore.max_tabs_undo: 1
browser.sessionstore.resume_from_crash: false
browser.sessionstore.upgradeBackup.latestBuildID: MOZ_BUILDID
browser.startup.homepage: about:blank
browser.startup.homepage_override.buildID: MOZ_BUILDID
browser.startup.homepage_override.mstone: 56.3
browser.tabs.closeWindowWithLastTab: false
browser.tabs.remote.autostart: false
browser.tabs.warnOnClose: false
browser.urlbar.lastSuggestionsPromptDate: 20180918
browser.urlbar.maxRichResults: 4
browser.urlbar.oneOffSearches: false
browser.urlbar.searchSuggestionsChoice: false
browser.urlbar.suggest.openpage: false
browser.urlbar.suggest.searches: false
browser.urlbar.timesBeforeHidingSuggestionsHint: 2
dom.event.clipboardevents.enabled: false
dom.webnotifications.enabled: false
extensions.lastAppVersion: 56.3
font.minimum-size.x-western: 9
font.name.sans-serif.x-western: Arial
font.name.serif.x-western: Arial
font.size.fixed.x-western: 11
font.size.variable.x-western: 14
gfx.crash-guard.d3d11layers.appVersion: 56.2.6
gfx.crash-guard.d3d11layers.deviceID: 0x0416
gfx.crash-guard.d3d11layers.driverVersion: 10.18.14.5057
gfx.crash-guard.d3d11layers.feature-d2d: true
gfx.crash-guard.d3d11layers.feature-d3d11: true
gfx.crash-guard.status.d3d11layers: 2
gfx.crash-guard.status.d3d11video: 2
image.animation_mode: once
layers.acceleration.disabled: true
layers.mlgpu.sanity-test-failed: false
media.autoplay.enabled: false
media.gmp-gmpopenh264.abi: x86_64-msvc-x64
media.gmp-gmpopenh264.lastUpdate: 1566494339
media.gmp-gmpopenh264.version: 1.8.1
media.gmp-manager.buildID: MOZ_BUILDID
media.gmp-manager.lastCheck: 1572615732
media.gmp.storage.version.observed: 1
media.hardware-video-decoding.failed: false
media.navigator.enabled: false
media.peerconnection.enabled: false
media.peerconnection.ice.proxy_only: true
network.cookie.cookieBehavior: 2
network.cookie.lifetimePolicy: 2
network.cookie.prefsMigrated: true
network.dns.disablePrefetch: true
network.http.sendRefererHeader: 1
network.http.speculative-parallel-limit: 0
network.predictor.cleaned-up: true
network.predictor.enabled: false
network.prefetch-next: false
places.database.lastMaintenance: 1572258637
places.history.expiration.transient_current_max_pages: 94467
plugin.disable_full_page_plugin_for_types: application/pdf
print.printer_HP_Deskjet_2540_series.print_bgcolor: false
print.printer_HP_Deskjet_2540_series.print_bgimages: false
print.printer_HP_Deskjet_2540_series.print_duplex: -437918235
print.printer_HP_Deskjet_2540_series.print_edge_bottom: 0
print.printer_HP_Deskjet_2540_series.print_edge_left: 0
print.printer_HP_Deskjet_2540_series.print_edge_right: 0
print.printer_HP_Deskjet_2540_series.print_edge_top: 0
print.printer_HP_Deskjet_2540_series.print_evenpages: true
print.printer_HP_Deskjet_2540_series.print_footercenter:
print.printer_HP_Deskjet_2540_series.print_footerleft: &PT
print.printer_HP_Deskjet_2540_series.print_footerright: &D
print.printer_HP_Deskjet_2540_series.print_headercenter:
print.printer_HP_Deskjet_2540_series.print_headerleft: &T
print.printer_HP_Deskjet_2540_series.print_headerright: &U
print.printer_HP_Deskjet_2540_series.print_in_color: true
print.printer_HP_Deskjet_2540_series.print_margin_bottom: 0.5
print.printer_HP_Deskjet_2540_series.print_margin_left: 0.5
print.printer_HP_Deskjet_2540_series.print_margin_right: 0.5
print.printer_HP_Deskjet_2540_series.print_margin_top: 0.5
print.printer_HP_Deskjet_2540_series.print_oddpages: true
print.printer_HP_Deskjet_2540_series.print_orientation: 0
print.printer_HP_Deskjet_2540_series.print_page_delay: 50
print.printer_HP_Deskjet_2540_series.print_paper_data: 9
print.printer_HP_Deskjet_2540_series.print_paper_height: -1.00
print.printer_HP_Deskjet_2540_series.print_paper_name:
print.printer_HP_Deskjet_2540_series.print_paper_size_unit: 1
print.printer_HP_Deskjet_2540_series.print_paper_width: -1.00
print.printer_HP_Deskjet_2540_series.print_resolution: 600
print.printer_HP_Deskjet_2540_series.print_reversed: false
print.printer_HP_Deskjet_2540_series.print_scaling: 1.00
print.printer_HP_Deskjet_2540_series.print_shrink_to_fit: true
print.printer_HP_Deskjet_2540_series.print_to_file: false
print.printer_HP_Deskjet_2540_series.print_unwriteable_margin_bottom: 0
print.printer_HP_Deskjet_2540_series.print_unwriteable_margin_left: 0
print.printer_HP_Deskjet_2540_series.print_unwriteable_margin_right: 0
print.printer_HP_Deskjet_2540_series.print_unwriteable_margin_top: 0
privacy.clearOnShutdown.downloads: false
privacy.clearOnShutdown.history: false
privacy.history.custom: true
privacy.sanitize.sanitizeOnShutdown: true
privacy.sanitize.timeSpan: 2
security.enterprise_roots.enabled: true
security.sandbox.content.tempDirSuffix: {e18dfb11-790d-4baf-9282-ea6fc926ef14}
storage.vacuum.last.index: 1
storage.vacuum.last.places.sqlite: 1570198983
ui.osk.debug.keyboardDisplayReason: IKPOS: Touch screen not found.
webgl.disabled: true

Important Locked Preferences
----------------------------

Places Database
---------------

JavaScript
----------

Incremental GC: true

Accessibility
-------------

Activated: false
Prevent Accessibility: 0
Accessible Handler Used: true

Library Versions
----------------

NSPR
Expected minimum version: 4.16
Version in use: 4.16

NSS
Expected minimum version: 3.34
Version in use: 3.34

NSSSMIME
Expected minimum version: 3.34
Version in use: 3.34

NSSSSL
Expected minimum version: 3.34
Version in use: 3.34

NSSUTIL
Expected minimum version: 3.34
Version in use: 3.34

Experimental Features
---------------------

Sandbox
-------

Content Process Sandbox Level: 1
Effective Content Process Sandbox Level: 1
